R/sortable_options.R

```diff
@@ -0,0 +1,98 @@
+# ' @param fallbackClass : "sortable-fallback",  // Class name for the cloned DOM Element when using forceFallback
+# ' @param fallbackOnBody : false,  // Appends the cloned DOM Element into the Document's Body
+# ' @param fallbackTolerance : 0, // Specify in pixels how far the mouse should move before it's considered as a drag.
+# ' @param dragoverBubble If set to true, the dragover event will bubble to parent sortables. \[`false`\]
+# ' @param removeCloneOnHide: true, // Remove the clone element when it is not showing, rather than just hiding it
+# ' @param emptyInsertThreshold: Number of pixels a mouse must be from empty sortable to insert drag element into it. \[`5`\]
+# ' @param setData undocumented on website
+# https://github.com/SortableJS/Sortable/tree/master/plugins/AutoScroll
+#' @param scrollSensitivity Number of pixels the mouse needs to be to an edge to start scrolling. \[`30`\]
+#' @param scrollSpeed Number of pixels for the speed of scrolling. \[`10`\]
+# ' @param onChoose,onUnchoose JS function called when an element is chosen or unchosen
+#' @param onStart,onEnd JS function called when an element dragging starts or ends
+#' @param onAdd JS function called when an element is dropped into the list from another list
+#' @param onUpdate JS function called when the sorting is changed within a list
+#' @param onSort JS function called by any change to the list (add / update / remove)
+#' @param onRemove JS function called when an element is removed from the list into another list
+#' @param onFilter JS function called when an attempt is made to drag a filtered element
+#' @param onMove JS function called when an item is moved in a list or between lists
+# ' @param onClone JS function that is called when creating a clone of an element
+# ' @param onChange JS function that is called when a dragging element changes position
+#' @param ... other params passed onto Sortable.js
+#' @seealso [https://github.com/sortablejs/Sortable/](https://github.com/sortablejs/Sortable/)
+#' @export
+sortable_options <- function(
+  # nolint start
+  group = NULL,
+  sort = NULL,
+  delay = NULL,
+  disabled = NULL,
+  animation = NULL,
+  handle = NULL,
+  filter = NULL,
+  draggable = NULL,
+  swapThreshold = NULL,
+  invertSwap = NULL,
+  direction = NULL,
+  scrollSensitivity = NULL,
+  scrollSpeed = NULL,
+  onStart = NULL,
+  onEnd = NULL,
+  onAdd = NULL,
+  onUpdate = NULL,
+  onSort = NULL,
+  onRemove = NULL,
+  onFilter = NULL,
+  onMove = NULL,
+  # nolint end
+  ...
+) {
+  extra_args <- list(...)
+
+  # get all names and values
+  args <- names(formals(sortable_options))
+  arg_vals <- mget(args[-length(args)], environment()) # remove last element (...)
+
+  # remove null values
+  is_null <- vapply(arg_vals, is.null, logical(1))
+  arg_vals <- arg_vals[!is_null]
+
+  # merge all args
+  ret <- append(arg_vals, extra_args)
+
+  class(ret) <- "sortable_options"
+  ret
+}
+
+
+
+assert_sortable_options <- function(x) {
```

Comment:
Let's use the `assertthat` package for assertions.

Reply:
For now, we should pass as it's only checking for a class.  If we do actual type checking, my go to is `checkmate` and I believe it should be done inside `sortable_options`.